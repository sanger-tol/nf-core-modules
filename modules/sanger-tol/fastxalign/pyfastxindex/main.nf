process FASTXALIGN_PYFASTXINDEX {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/63/637ecb387eaafe0b1689e3e32c5eda589e016cfd46c482946425181f69f0733e/data' :
        'community.wave.seqera.io/library/htslib_minimap2_pyfastx_samtools_click:bfd8f60cc27aa6d6' }"

    input:
    tuple val(meta), path(fastx)

    output:
    tuple val(meta), path("*.fxi"), stdout, emit: index
    path "versions.yml"                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    // WARNING: This module includes the slice_fasta.py script as a module binary in
    // ${moduleDir}/resources/usr/bin/slice_fasta.py. To use this module, you will
    // either have to copy this file to ${projectDir}/bin or set the option
    // nextflow.enable.moduleBinaries = true
    // in your nextflow.config file.
    def args       = task.ext.args  ?: ''
    """
    slice_fasta.py index \\
        ${fastx} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        slice_fasta.py: \$(slice_fasta.py --version)
    END_VERSIONS
    """

    stub:
    """
    touch ${fastx}.fxi
    ## output dummy count to stdout
    echo -n 100

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        slice_fasta.py: \$(slice_fasta.py --version)
    END_VERSIONS
    """
}
