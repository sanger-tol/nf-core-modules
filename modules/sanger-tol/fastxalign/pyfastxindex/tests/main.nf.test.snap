{
    "Undibacterium_unclassified - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "HiFi.reads.fasta.fxi:md5,3e9c4aa709ec1b185045bfd44d07db22",
                        "930"
                    ]
                ],
                "1": [
                    "versions.yml:md5,df71a9c1c15afae42d50c6150e13df66"
                ],
                "index": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "HiFi.reads.fasta.fxi:md5,3e9c4aa709ec1b185045bfd44d07db22",
                        "930"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,df71a9c1c15afae42d50c6150e13df66"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-20T14:54:55.456528"
    },
    "Undibacterium_unclassified - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "HiFi.reads.fasta.fxi:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "100"
                    ]
                ],
                "1": [
                    "versions.yml:md5,df71a9c1c15afae42d50c6150e13df66"
                ],
                "index": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "HiFi.reads.fasta.fxi:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "100"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,df71a9c1c15afae42d50c6150e13df66"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-20T14:55:08.398709"
    }
}